docs/Using Randomness/v1/get-a-random-number.md

```diff
@@ -0,0 +1,102 @@
+---
+layout: nodes.liquid
+section: smartContract
+date: Last Modified
+title: "Get a Random Number"
+permalink: "docs/get-a-random-number/v1/"
+whatsnext: {"API Reference":"/docs/chainlink-vrf-api-reference/v1/", "Contract Addresses":"/docs/vrf-contracts/v1/"}
+metadata: 
+  description: "How to generate a random number inside a smart contract using Chainlink VRF."
+  image: 
+    0: "/files/OpenGraph_V3.png"
+---
+This page explains how to get a random number inside a smart contract using Chainlink VRF.
+
+
+
+# Random Number Consumer
+
+Chainlink VRF follows the [Request & Receive Data](../../request-and-receive-data/) cycle. To consume randomness, your contract should inherit from <a href="https://github.com/smartcontractkit/chainlink/blob/master/evm-contracts/src/v0.6/VRFConsumerBase.sol" target="_blank">`VRFConsumerBase`</a> and define two required functions
+
+1. `requestRandomness`, which makes the initial request for randomness.
+2. `fulfillRandomness`, which is the function that receives and does something with verified randomness.
+
+The contract should own enough LINK to pay the specified fee. The beginner walkthrough explains how to [fund your contract](../../fund-your-contract/).
+
+Note, the below values have to be configured correctly for VRF requests to work. You can find the respective values for your network in the [VRF Contracts page](../../vrf-contracts/v1).
+
+- `LINK Token` - LINK token address on the corresponding network (Ethereum, Polygon, BSC, etc)
+- `VRF Coordinator` - address of the Chainlink VRF Coordinator
+- `Key Hash` - public key against which randomness is generated
+- `Fee` - fee required to fulfill a VRF request
+
+> 🚧 Security Considerations
+>
+> Be sure to look your contract over with [these security considerations](../../vrf-security-considerations/v1/) in mind!
+
+>❗️ Remember to fund your contract with LINK!
+>
+> Requesting randomness will fail unless your deployed contract has enough LINK to pay for it. **Learn how to [Acquire testnet LINK](../../acquire-link/) and [Fund your contract](../../fund-your-contract/)**.
+
+<div class="remix-callout">
+    <a href="https://remix.ethereum.org/#version=soljson-v0.6.6+commit.6c089d02.js&optimize=false&evmVersion=null&gist=f47e4eae5f2ffa7868ef4ecd5bda9044&runs=200" target="_blank" class="cl-button--ghost solidity-tracked">Deploy this contract using Remix ↗</a>
```

Comment:
We switched from Gists to use a samples directory. Add a copy of the code here: https://github.com/smartcontractkit/documentation/tree/main/_includes/samples

Reply:
Will do. Is there a way I could just point it directly to the core repo examples i.e. https://github.com/smartcontractkit/chainlink/blob/b56b40353b0df29f058e3d2d79d55c1caaf60031/contracts/src/v0.8/tests/VRFSingleConsumerExample.sol then have them in one place?

Reply:
Right now we pull our examples from _includes/samples and serve them out of the docs.chain.link/samples directory.  Here is the passthrough for that: https://github.com/smartcontractkit/documentation/blob/0fe5549c0eba0e7f641d4e80940d85ba2d5e62c3/.eleventy.js#L16

Remix is capable of pulling in example code from public GitHub repositories, but we want to serve the examples from our site if we can. We might be able to do a copy from the core repo at build time, but we would need to put it on a schedule so it picks up changes. I'll look into the best way of doing that. I definitely prefer to single-source examples.